elastic/kibana
# [Index Management] Remove `_cat/indices` API from server code

Fixes https://github.com/elastic/kibana/issues/57286
Fixes https://github.com/elastic/kibana/issues/90565 (skipped api integration test)

## Summary
This PR refactors server side code of indices list route so that it doesn't rely on `_cat/indices` API. There should not be any visual or functional regressions in the Indices tab of Index Management. 

## How to test 
Note: We need to wait until https://github.com/elastic/elasticsearch/pull/81954 is in the ES snapshot build for CI tests to pass and to be able to run `yarn es snapshot` and test locally. 
1. Start ES with `yarn es snapshot` 
2. Start Kibana with `yarn start`
3. Navigate to Index Management -> Indices and check that indices information in the table is correct (Toggle "include hidden indices" to show more indices)
3. Create an index in Dev Tools Console with 
```
POST /test_index/_doc
{
  "test": "test"
}
```
4.  In Index Management, open "test_index" details flyout and change its settings: number of shards and replicas, open/close the index and check that this information is correctly updated in the table. 
5. Add more documents to "test_index" by running the same `_doc` request a couple of times
```
POST /test_index/_doc
{
  "test": "test"
}
```
Check that index size and docs number is updated in the table.
